Fix table ref warn

MMFane opened the pull request:
# What Was Changed

- Finally fixing some console pollution I introduced back when we added sticky columns to `<TableModule />`
- We now only show the 'Table doesn't have a ref...' warning if some sticky columns are set
    - This will also prevent unnecessary spacing calculations being run when there are no sticky cols
- We also now only show this warning in Dev

# Screenshots

## Before

This warning appeared for every table on every page that did not have a ref, regardless of whether it needed sticky cols

<img width="1048" alt="Screenshot 2023-08-04 at 4 52 22 PM" src="https://github.com/lifeomic/chroma-react/assets/5824697/8cb0315c-3367-4831-8d1a-39dd32211e85">

## After

Now nothing should appear for non-sticky tables, or in Prod

shawnzhu commented on src/components/TableModule/TableModule.tsx:
```diff
@@ -374,8 +378,9 @@ export const TableModule = React.memo(
         } else {
-          console.warn(
-            "Table's forwardRef is null, please set it if you want the sticky cell's left value to be set correctly"
+          warning(
+            process.env.NODE_ENV === 'development',
+            "Chroma Warning: Table's forwardRef is null, please set it in order to set the sticky cell's left value correctly"
```
`process` object is only available in node.js runtime. it will report error when interpreting it in a browser. So it could just keep these lines as they were

MMFane replied:
OK - I was modeling this warning off the one on line 291 - should I adjust that to a plain console.warn as well? I think this util is common around Chrōma, so I'm wondering if we have a larger issue to address.

shawnzhu replied:
just use `console.warn()` like what it was. I don't think you need to change these lines except your fix

swain replied:
(I also agree with Shawn's immediate feedback: I'd suggest skipping the `process.env` check and just doing `console.warn`)

MMFane replied:
Cool, thanks for the feedback, guys!

epeters3 commented:
> Finally fixing some console pollution I introduced back when we added sticky columns to <TableModule />

So you're the celebrity that introduced sticky columns! I've been enjoying that feature in our Patient ML Runs table.

MMFane commented:
> > Finally fixing some console pollution I introduced back when we added sticky columns to
> 
> So you're the celebrity that introduced sticky columns! I've been enjoying that feature in our Patient ML Runs table.

Yay! Glad to hear it's been helpful! ✨✨✨
